### PR TITLE
Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by luceos**
updates our PR template to clarify a few things

fun thing having to fill out the PR template when updating the PR template..

**Fixes nothing**

**Changes proposed in this pull request:**
- PR template

**Reviewers should focus on:**
- required steps before sending in a PR
- the comment description

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
 
 **Screenshot**
 <!-- include an image of the most relevant user-facing change, if any -->
+
+**Confirmed**
+
+- [ ] for visual or frontend changes that the changes work as intended via visual confirmation on a local flarum installation.
+- [ ] for php changes that the unit tests are all AOK (green) by running them locally (`php vendor/bin/phpunit`).
+
+**Required changes:**
```
[the remark this answers is not in the document]
Comment (luceos) [reply .github/PULL_REQUEST_TEMPLATE.md]: line 21 is, the lines above that are meant as checks/requirements

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
-<!-- fill this out -->
+<!-- fill this out, ask for feedback on specific changes you are unsure about -->
 
 **Screenshot**
 <!-- include an image of the most relevant user-facing change, if any -->
+
+**Confirmed**
+
+- [ ] for visual or frontend changes that the changes work as intended via visual confirmation on a local flarum installation.
+- [ ] for php changes that the unit tests are all AOK (green) by running them locally (`php vendor/bin/phpunit`).
```
[the remark this answers is not in the document]
Comment (luceos) [reply .github/PULL_REQUEST_TEMPLATE.md]: Agreed.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (franzliedke) [reply .github/PULL_REQUEST_TEMPLATE.md]: What I'm saying is:
- "Docs" sounds like more of a TODO for later,
- "Core extensions" seems to be the place for links to related PRs.

Or should both contain links to related PRs?

[the remark this answers is not in the document]
Comment (tobscure) [reply .github/PULL_REQUEST_TEMPLATE.md]: Ideally docs should be PR'd at the same time (at least once we have a good foundation of docs!)

[the remark this answers is not in the document]
Comment (luceos) [reply .github/PULL_REQUEST_TEMPLATE.md]: Nice one.

**Comment by luceos** [comment]
@franzliedke i leave the final review and merge to you. I've updated the top description a little. We can iterate later if we feel it could use more love. The suggestions were integrated.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobscure) [reply .github/PULL_REQUEST_TEMPLATE.md]: Can we also add "updated changelog" as a checklist item. From now on I'd like to keep a CHANGELOG.md in the root.

**Comment by tobscure** [comment]
Can we also add "updated changelog" as a checklist item. From now on I'd like to keep a CHANGELOG.md in the root.

**Comment by luceos** [comment]
@tobscure good point.

@flarum/core with these kind of PRs, just use the inline github edit to add your suggestions (if they seem obvious)?

**Comment by franzliedke** [comment]
Regarding the changelog: This is just asking for lots of conflicts between PRs, especially if they sit around for a while (ahem).

**Comment by luceos** [comment]
Not if they are added to a heading called `# next`? We can create such a thing per default.

**Comment by franzliedke** [comment]
@luceos But everyone will add their change at the bottom of that file, meaning we will have multiple inserts. Still a conflict. :confused:

**Comment by tobscure** [comment]
True. Relevant discussion:

https://github.com/isaacs/github/issues/487
https://github.com/olivierlacan/keep-a-changelog/issues/56

**Comment by luceos** [comment]
We can still pull the changes from the PR description once we do a release. Doing a release has been a manual action so far anyway.

**Comment by luceos** [comment]
@tobscure how about a release issue for the upcoming release named Changelog?

**Comment by franzliedke** [comment]
What is keeping us from using the release milestone then? "Some changes are not related to issues"?
What about a GitHub project per release then, where we can add any small change that we want to keep track of as one of these low-fidelity "notes"? That way, we don't have to worry about the order of changes (and thus, conflicts), but still have all the changes that we want to track in one place.

**Comment by franzliedke** [comment]
Also fun, but probably overdoing it a bit:
https://about.gitlab.com/2018/07/03/solving-gitlabs-changelog-conflict-crisis/

:sweat_smile:

**Comment by datitisev** [comment]
This is why having a Commit template such as angular’s can be nice when creating a CHANGELOG automatically or manually.

**Comment by luceos** [comment]
@datitisev link? example?

**Comment by datitisev** [comment]
Angular CONTRIBUTING.md: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md
Angular Commit Guidelines: https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits

Then their changelog is generated with `cz-conventional-changelog`, I think (uses [Commitizen](https://www.npmjs.com/package/commitizen) to make writing commits easier)
image

**Pull request merged**